File: icpc 초보/2022-겨울-스터디-초급/7회차 필수문제 코드/cpp/1802.cc
#include <bits/stdc++.h>
using namespace std;

int tc;
string s;

int f(int l, int r) {
	if (l == r) return 1;
	int m = (l + r) / 2, lenHalf = (r - l + 1) / 2;
	if (!(f(l, m - 1) & f(m + 1, r))) return 0;
	for (int i = 0; i < lenHalf; i++)
		if (1 - (s[l + i] - '0') != s[r - i] - '0') return 0;
	return 1;
}

int main() {
#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
#endif
	ios_base::sync_with_stdio(false); cin.tie(0);
	cin >> tc;
	while (tc--) {
		cin >> s;
		int res = f(0, (int)s.size() - 1);
		if (res) cout << "YES\n";
		else cout << "NO\n";
	}
	return 0;
}

File: icpc 초보/2022-겨울-스터디-초급/7회차 필수문제 코드/python/1802.py
def solve(s):
    if len(s) == 1:
        return True
    for i in range(len(s) // 2):
        if int(s[i]) + int(s[len(s) - 1 - i]) != 1:
            return False

    return solve(s[ : len(s) // 2]) and solve(s[len(s) // 2 + 1: ])
t = int(input())

for tc in range(t):
    s = input()

    if (solve(s)):
        print("YES")
    else:
        print("NO")